subsys/shell/modules/devmem_service.c

```diff
@@ -1,12 +1,140 @@
 /*
  * Copyright (c) 2020 Intel Corporation
+ * Copyright (c) 2021 Antmicro <www.antmicro.com>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <stdlib.h>
 #include <device.h>
 #include <shell/shell.h>
-#include <stdlib.h>
+#include <sys/byteorder.h>
+
+static inline bool is_ascii(uint8_t data)
+{
+	if ((data >= 0x30 && data <= 0x39) || (data >= 0x61 && data <= 0x66) ||
+	    (data >= 0x41 && data <= 0x46)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+static unsigned char *bytes;
+static uint32_t *data;
+static int sum;
+static int chunk_element;
+static char chunk[2];
+static bool littleendian;
+
+#define CHAR_CAN 0x18
+#define CHAR_DC1 0x11
+
+static int set_bypass(const struct shell *sh, shell_bypass_cb_t bypass)
+{
+	static bool in_use;
+
+	if (bypass && in_use) {
+		shell_error(sh, "devmem load supports setting bypass on a single instance.");
+
+		return -EBUSY;
+	}
+
+	in_use = !in_use;
+	if (in_use) {
+		shell_print(sh, "Loading...\npress ctrl-x ctrl-q to escape");
+		in_use = true;
+	}
+
+	shell_set_bypass(sh, bypass);
+
+	return 0;
+}
+
+static void bypass_cb(const struct shell *sh, uint8_t *recv, size_t len)
+{
+	bool escape = false;
+	static uint8_t tail;
+	uint8_t byte;
+
+	if (tail == CHAR_CAN && recv[0] == CHAR_DC1) {
+		escape = true;
+	} else {
+		for (int i = 0; i < (len - 1); i++) {
+			if (recv[i] == CHAR_CAN && recv[i + 1] == CHAR_DC1) {
+				escape = true;
+				break;
+			}
+		}
+	}
+
+	if (escape) {
+		shell_print(sh, "Number of bytes read: %d", sum);
+		set_bypass(sh, NULL);
+
+		if (!littleendian) {
+			while (sum > 4) {
+				*data = __bswap_32(*data);
+				data++;
+				sum = sum - 4;
+			}
+			if (sum % 4 == 0) {
+				*data = __bswap_32(*data);
+			} else if (sum % 4 == 2) {
+				*data = __bswap_16(*data);
+			} else if (sum % 4 == 3) {
+				*data = __bswap_24(*data);
+			}
+		}
+		return;
+	}
+
+	tail = recv[len - 1];
+
+	if (is_ascii(*recv)) {
+		chunk[chunk_element] = *recv;
+		chunk_element++;
+	}
+
+	if (chunk_element == 2) {
+		byte = (uint8_t)strtoul(chunk, NULL, 16);
+		*bytes = byte;
+		bytes++;
+		sum++;
+		chunk_element = 0;
+	}
+}
+
+static int cmd_load(const struct shell *sh, size_t argc, char **argv)
+{
+	littleendian = false;
+	char *arg;
+
+	chunk_element = 0;
+	sum = 0;
+
+	while (argc >= 2) {
+		arg = argv[1] + (!strncmp(argv[1], "--", 2) && argv[1][2]);
+		if (!strncmp(arg, "-e", 2))
+			littleendian = true;
+		else if (!strcmp(arg, "--")) {
+			argv++;
+			argc--;
+			break;
+		} else if (arg[0] == '-' && arg[1])
+			shell_print(sh, "Unknown option \"%s\"", arg);
+		else
+			break;
```

Comment:
I'm not sure why checkpatch didn't catch this, but the last `else if` and `else` should probably have braces added to them. From https://docs.zephyrproject.org/latest/contribute/index.html#coding-style:

> Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks

Reply:
I've added braces

Reply:
I think that braces are still missing in the `else if` (line 120 after your changes). :)

Reply:
Sorry about that, now all if's have braces.